src/neo4j_graphrag/embeddings/openai.py

```diff
@@ -64,7 +64,20 @@ def embed_query(self, text: str, **kwargs: Any) -> list[float]:
         return response.data[0].embedding
 
 
-class AzureOpenAIEmbeddings(OpenAIEmbeddings):
+class AzureOpenAIEmbeddings(BaseOpenAIEmbeddings):
     def __init__(self, model: str = "text-embedding-ada-002", **kwargs: Any) -> None:
         super().__init__(model, **kwargs)
         self.openai_client = openai.AzureOpenAI(**kwargs)
+
+    def embed_query(self, text: str, **kwargs: Any) -> list[float]:
```

Comment:
Shall we move this method to the base class? If it's exactly the same one for both OpenAI and AzureOpenAI

Reply:
There would be a type hint warning as `self.opena_client` isn't initialized. I tried initializing `self.open_client = None`  but 

```python
self.openai_client.embeddings.create(
            input=text, model=self.model, **kwargs
        )
```

would raise an error with mypy

Reply:
Ok I see. We can maybe use a similar approach to @jonbesga in #175 for the OpenAI LLM class: 

```
if TYPE_CHECKING:
    import openai
```

then, in the base class:
```
    client: openai.OpenAI
```

and I think AzureOpenAI inherits from OpenAI, so it mypy should be happy.

Reply:
We can also wait until the type import fixes PR is merged before harmonizing, I'm OK with that solution too.

Reply:
https://github.com/neo4j/neo4j-graphrag-python/pull/203/commits/4e86acbefbb530ac7bf61276a3b25ae6d402f7c2